### PR TITLE
Stringorslice and Command refactor

### PR DESCRIPTION
**Pull request by zh-lab**
I saw TODO and FIXME above type Stringorslice and Command, so I try to work on it.

Signed-off-by: Zhiheng Huang <zh-@outlook.com>

### CODE REVIEW

File: project/types_yaml.go [hunk 1]
```diff
@@ -41,81 +39,47 @@ func (s *Stringorslice) UnmarshalYAML(tag string, value interface{}) error {
...
 }
 
 // Command represents a docker command, can be a string or an array of strings.
-// FIXME why not use Stringorslice (type Command struct { Stringorslice }
-type Command struct {
-	parts []string
-}
-
-// MarshalYAML implements the Marshaller interface.
-func (s Command) MarshalYAML() (tag string, value interface{}, err error) {
```
Comment (vdemeester) [inline project/types_yaml.go]: I'm not sure we should remove this one though.

**Comment by vdemeester** [comment]
@zh-lab note that this has been move in current `docker/docker` project (see https://github.com/docker/docker/commit/f9b857a200696b07b67e6a7f94ede32487f5649d)

File: project/types_yaml.go (hunk 1 above)
Comment (zh-lab) [inline project/types_yaml.go]: Stringorslice's MarshalYAML will be promoted here.
Could we use that?

**Comment by zh-lab** [comment]
@vdemeester Thanks for reminding me that StrSlice was moved! :flushed:
So we should keep Stringorslice untouched now?

**Comment by vdemeester** [comment]
@zh-lab I would think so, I'm waiting for more API stabilization to vendor `docker/docker` again ; and to make it easier I would let it untouched :stuck_out_tongue_closed_eyes:

**Comment by vdemeester** [comment]
Closing this one (see https://github.com/docker/libcompose/pull/125#issuecomment-167236482). Thanks @zh-lab, will ping you when bumping docker 1.10 to see what we can do about this `TODO` :wink:.

**Pull request closed without merging**